common/txmgr/priority_queue.go

```diff
@@ -0,0 +1,81 @@
+package txmgr
+
+import (
+	"container/heap"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+// TxPriorityQueue is a priority queue of transactions prioritized by creation time. The oldest transaction is at the front of the queue.
+type TxPriorityQueue[
+	CHAIN_ID types.ID,
+	ADDR, TX_HASH, BLOCK_HASH types.Hashable,
+	R txmgrtypes.ChainReceipt[TX_HASH, BLOCK_HASH],
+	SEQ types.Sequence,
+	FEE feetypes.Fee,
+] struct {
+	*txmgrtypes.PriorityQueue[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]
```

Comment:
You don't need to make the change, but just wondering out aloud.
The only pattern we have need for, is add items to end of queue, and remove items from the beginning.
And at times, search an item in this list, and remove it.

Could we not be better served instead by a doubly liked list data structure?
In Linked List, insertions and delete from head would be O(1), and deletions by id would be O(n).

Just a thought for future. Not to do this right away